### PR TITLE
[CHK-2177] refactor(api): move `PATCH /wallet/{id}` -> `PUT /wallet/{id}/services`

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

 * refactor: move `PATCH /wallet/{id}` -> `PUT /wallet/{id}/services`
 * In case of partial wallet services update, return a `409 Conflict` with both the services that were updated and the services that generated the conflict

#### Motivation and Context

For the HTTP method change: requested by our users :)
For the status code: it can give more information to downstream users on what to do in case of not being able to update wallet services as desired

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/domain/services/ServiceStatus.kt [hunk 1]
```diff
@@ -3,5 +3,29 @@ package it.pagopa.wallet.domain.services
...
+                       Requested
+                        E  D  I
+                 G     ┌──┬──┬──┐
+                 l   E │OK│OK│NO│
+                 o     ├──┼──┼──┤
+                 b   D │NO│OK│NO│
+                 a     ├──┼──┼──┤
+                 l   I │NO│OK│OK│
+                       └──┴──┴──┘
+            */
```
Comment (ciuffagianluca) [inline src/main/kotlin/it/pagopa/wallet/domain/services/ServiceStatus.kt:22]: ❤️ love this

File: src/main/kotlin/it/pagopa/wallet/services/WalletService.kt [hunk 2]
```diff
@@ -529,36 +518,71 @@ class WalletService(
...
+                            mutableMapOf<ServiceName, ServiceStatus>()
+                        )
+                    ) {
+                        (servicesUpdatedSuccessfully, servicesWithUpdateFailed),
+                        (service, serviceName, requestedStatus) ->
+                        val serviceGlobalStatus = ServiceStatus.valueOf(service.status)
+                        val walletApplication = walletApplications[serviceName.name]
+
+                        if (
+                            ServiceStatus.canChangeToStatus(
```
Comment (ciuffagianluca) [inline src/main/kotlin/it/pagopa/wallet/services/WalletService.kt:550]: Coverage for all pr code is really good. But You could improve adding specific test for this method.

**Review by ciuffagianluca — COMMENTED** [review]
I updated the base branch but there are some spotless to be fixed 🙂

File: src/main/kotlin/it/pagopa/wallet/services/WalletService.kt (hunk 2 above)
Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/wallet/services/WalletService.kt:549]: Fixed with d94e391

**Review by ciuffagianluca — APPROVED** [review]
Well done!

**Pull request merged**